graylog2-web-interface/src/components/features/Feature.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import useFeature from 'hooks/useFeature';
+
+type Props = {
```

Comment:
I found this [article](https://blog.logrocket.com/comparing-typescript-and-proptypes-in-react-applications/) yesterday. It is possible to infer props and remove the duplicate type definition but we have to install @types/prop-types as a dev dependency to use it. It is only a suggestion.

Reply:
Yes, I totally agree. It's a broader topic.

Reply:
The issue is that prop types are less expressive than TypeScript. IMO we would want it the other way around: Generating prop types from TypeScript typings. There is https://github.com/milesj/babel-plugin-typescript-to-proptypes, but it does not look too good (no release for a year, some rather unfortunate issues we would run into like https://github.com/milesj/babel-plugin-typescript-to-proptypes/issues/45), so this is still an unsolved question.

Our general idea is this:

  - Always use TypeScript types (when there is no real good argument against that)
  - Use prop types for components which are reused across different contexts

This should strike a good compromise in most cases.